src/clj_time/core.clj

```diff
@@ -307,99 +311,102 @@
   [dt & dts]
   (reduce #(if (after? %1 %2) %1 %2) dt dts))
 
-(defn ^org.joda.time.DateTime date-time
+(defn date-time
   "Constructs and returns a new DateTime in UTC.
    Specify the year, month of year, day of month, hour of day, minute of hour,
    second of minute, and millisecond of second. Note that month and day are
    1-indexed while hour, second, minute, and millis are 0-indexed.
    Any number of least-significant components can be ommited, in which case
    they will default to 1 or 0 as appropriate."
-  ([year]
+  (^org.joda.time.DateTime [year]
    (date-time year 1 1 0 0 0 0))
-  ([year month]
+  (^org.joda.time.DateTime [year month]
    (date-time year month 1 0 0 0 0))
-  ([year month day]
+  (^org.joda.time.DateTime [year month day]
    (date-time year month day 0 0 0 0))
-  ([year month day hour]
+  (^org.joda.time.DateTime [year month day hour]
    (date-time year month day hour 0 0 0))
-  ([year month day hour minute]
+  (^org.joda.time.DateTime [year month day hour minute]
    (date-time year month day hour minute 0 0))
-  ([year month day hour minute second]
+  (^org.joda.time.DateTime [year month day hour minute second]
    (date-time year month day hour minute second 0))
-  ([^Integer year ^Integer month ^Integer day ^Integer hour
-    ^Integer minute ^Integer second ^Integer millis]
+  (^org.joda.time.DateTime [^Integer year ^Integer month ^Integer day ^Integer hour
```

Comment:
This repeated type hint pattern may look ugly, but as pointed out, it is the only officially documented way of annotating ret vals, and the only that doesn't perform an extra, spurious annotation.